src/hankel.jl

```diff
@@ -0,0 +1,116 @@
+# Hankel
+struct Hankel{T<:Number} <: AbstractMatrix{T}
+    v::AbstractVector{T}
+    s::Dims{2} # size
```

Comment:
s should probably also be templated

Reply:
Is it for abstract axes like quasimatrices?

Reply:
No for the infinite case